analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd

```diff
@@ -211,22 +227,56 @@ any(duplicated(rna_df$sample_id))
 ```
 What are they?
 
+Checking for duplicate rna sample_ids
+
 ```{r}
 histologies_df %>%
   filter(sample_id %in% rna_df$sample_id[which(duplicated(rna_df$sample_id))],
          experimental_strategy == "RNA-Seq") %>%
   group_by(sample_id) %>%
-  summarize(RNA_library = paste(RNA_library, collapse = ", "))
+  summarize(RNA_library = paste(RNA_library, collapse = ", "),
+            composition =  paste(composition, collapse = ", "))
 ```
 
-There are 2 instances where there are two stranded biospecimen IDs that map to the same `sample_id`.
+The duplicates are mostly from multiple Solid tumor samples per sample_id or there are non-tumor ( derived cell-line ) samples for sample_id 
+
+
+Checking if adding cell-line composition type can uniquely match DNA to RNA
+
+```{r}
+
+cell_line_composition <- read_tsv("input/cell-line-composition.tsv",col_types = 
+                                    readr::cols( aliquot_id  = readr::col_character()))
+
+cell_line_composition %>%
+  left_join(histologies_df) %>%
+  group_by(sample_id,experimental_strategy) %>%
+  summarize(cell_line_composition = paste(cell_line_composition, collapse = ", "),
+            composition =  paste(composition, collapse = ", "))
+
+```
+`cell_line_composition` seems to be able to differentiate each cell line so let's add the column to merged dna and rna files
```

Comment:
It's not clear to me, without additional explanation, how the table demonstrates that `cell_line_composition` can differentiate each cell line. What would the table look like if that wasn't the case?

Reply:
hmm @jaclyn-taroni @jharenza  I was making that comment looking at the table displayed just before:
<img width="849" alt="Screen Shot 2020-10-12 at 10 35 20 AM" src="https://user-images.githubusercontent.com/34580719/95758623-b3f84a80-0c76-11eb-8165-719f18644482.png">


So looking at the table for example 7316-1746 RNA-Seq sample using the cell-line composition `A` or `S` we will be able to differentiate the multiple cell-line issue . 

sample_id<chr> | experimental_strategy<chr> | cell_line_composition<chr> | composition
-- | -- | -- | --
7316-1746 | RNA-Seq | A, S | Derived Cell Line, Derived Cell Line

Should I remove the chunk in DNA and RNA if it doesn't fit the code logic ?

Reply:
If it could _not_ differentiate, would that result look like:

sample_id | experimental_strategy<chr> | cell_line_composition<chr> | composition
-- | -- | -- | --
7316-1746 | RNA-Seq | A | Derived Cell Line, Derived Cell Line

Reply:
> If it could not differentiate

the results would look like : 

sample_id<chr> | experimental_strategy<chr> | cell_line_composition<chr> | composition
-- | -- | -- | --
7316-1746 | RNA-Seq | A, A | Derived Cell Line, Derived Cell Line

^ from here we can't differentiate the the 2 cell-lines with the same sample_id

Reply:
Ah, I see, no `unique()` step. That makes sense.